torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1791,6 +1791,7 @@ def error_inputs_margin_ranking_loss(op, device, **kwargs):
                      error_regex='margin_ranking_loss : All input tensors should')
 
 def sample_inputs_new_fns(self, device, dtype, requires_grad, *, is_strided=False, **kwargs):
+    other_dtype = torch.half if torch.backends.mps.is_available() else torch.double
```

Comment:
is there a better variable we can use to probe for float64 support?

Reply:
No, not really. Or at least I can't think of any. Also, this test perhaps should be further extended to pick different dtype if input dtype is torch.double/`torch.half`